src/transformers/generation/candidate_generator.py

```diff
@@ -367,29 +367,20 @@ def _crop_past_key_values(model, past_key_values, maximum_length):
     """Crops the past key values up to a certain maximum length."""
     new_past = []
     if model.config.is_encoder_decoder:
-        for idx in range(len(past_key_values)):
-            new_past.append(
-                (
-                    past_key_values[idx][0][:, :, :maximum_length, :],
-                    past_key_values[idx][1][:, :, :maximum_length, :],
-                    past_key_values[idx][2],
-                    past_key_values[idx][3],
-                )
-            )
-        past_key_values = tuple(new_past)
-    # bloom is special
-    elif "bloom" in model.__class__.__name__.lower() or (
-        model.config.architectures is not None and "bloom" in model.config.architectures[0].lower()
-    ):
-        for idx in range(len(past_key_values)):
-            new_past.append(
-                (
-                    past_key_values[idx][0][:, :, :maximum_length],
-                    past_key_values[idx][1][:, :maximum_length, :],
+        if isinstance(past_key_values[0], DynamicCache):
+            past_key_values[0].crop(maximum_length)
```

Comment:
This is related to the Whisper PR, correct? If so, this change should come in that PR, and not on this one :)

(In fact, I'm not very happy with `past_key_values[0]` -- see the Whisper PR 😛 )